onnx/backend/test/cpp/driver/test_driver.cc

```diff
@@ -199,25 +198,62 @@ ResolvedTestCase LoadSingleTestCase(const UnsolvedTestCase& t) {
   st.test_case_name_ = t.test_case_name_;
   ONNX_NAMESPACE::ParseProtoFromBytes(
       &st.model_, raw_model.c_str(), raw_model.size());
-
+  int test_data_counter = 0;
+  std::string tensor_str = "Tensor";
+  std::string sequence_str = "Sequence";
+  std::string map_str = "Map";
   for (auto& test_data : t.test_data_) {
     ResolvedTestData proto_test_data;
-
     for (auto& input_file : test_data.input_filenames_) {
       std::string input_data;
+      ONNX_NAMESPACE::ValueInfoProto input_info;
       LoadSingleFile(input_file, input_data);
-      ONNX_NAMESPACE::TensorProto input_proto;
-      ONNX_NAMESPACE::ParseProtoFromBytes(
-          &input_proto, input_data.c_str(), input_data.size());
-      proto_test_data.inputs_.emplace_back(std::move(input_proto));
+      input_info = st.model_.graph().input(test_data_counter);
+      if(typeid(input_info.type()) == typeid(ONNX_NAMESPACE::TypeProto::Tensor)) {
+        ONNX_NAMESPACE::TensorProto input_proto;
+        ONNX_NAMESPACE::ParseProtoFromBytes(
+            &input_proto, input_data.c_str(), input_data.size());
+        proto_test_data.inputs_.emplace_back(std::move(input_proto));
+      }
+      else if(typeid(input_info.type()) == typeid(ONNX_NAMESPACE::TypeProto::Sequence)) {
+        ONNX_NAMESPACE::SequenceProto input_proto;
+        ONNX_NAMESPACE::ParseProtoFromBytes(
+            &input_proto, input_data.c_str(), input_data.size());
+        proto_test_data.seq_inputs_.emplace_back(std::move(input_proto));
+      }
+      else if(typeid(input_info.type()) == typeid(ONNX_NAMESPACE::TypeProto::Map)) {
+        ONNX_NAMESPACE::MapProto input_proto;
+        ONNX_NAMESPACE::ParseProtoFromBytes(
+            &input_proto, input_data.c_str(), input_data.size());
+        proto_test_data.map_inputs_.emplace_back(std::move(input_proto));
+      }
+      test_data_counter++;
     }
+    test_data_counter = 0;
     for (auto& output_file : test_data.output_filenames_) {
-      std::string output_data = "";
+      std::string output_data;
+      ONNX_NAMESPACE::ValueInfoProto output_info;
+      output_info = st.model_.graph().output(test_data_counter);
       LoadSingleFile(output_file, output_data);
-      ONNX_NAMESPACE::TensorProto output_proto;
-      ONNX_NAMESPACE::ParseProtoFromBytes(
-          &output_proto, output_data.c_str(), output_data.size());
-      proto_test_data.outputs_.emplace_back(std::move(output_proto));
+      if(typeid(output_info.type()) == typeid(ONNX_NAMESPACE::TypeProto::Tensor)) {
```

Comment:
I don't think we should use "typeid" here. I doubt if it would work. Protobuf has its own encoding for "oneof" : see https://developers.google.com/protocol-buffers/docs/reference/cpp-generated#oneof-embedded-message-fields ... I think we should probably be using output_info.type().has_tensor_type() ... similarly, has_sequence_type() and has_map_type() for below.

Reply:
It would be good to validate if the right branch is executing here. I suspect none of these branches may be executing, but maybe I am wrong.

Reply:
Updated to use .has_tensor_type() and related functions!

Reply:
After testing with the new methods, it is hitting those if branches. :)